build/Common.props

```diff
@@ -54,6 +54,10 @@
     <PackageReference Include="StyleCop.Analyzers" Version="$(StyleCopAnalyzersPkgVer)" Condition="'$(SkipAnalysis)'!='true'" PrivateAssets="All" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(IncludeSharedGuardSource)'=='true'">
+    <Compile Include="$(RepoRoot)\src\Shared\Guard.cs" Link="Includes\Guard.cs" />
```

Comment:
I don't see much value in extracting this into a variable. It's a single compile include item. I think extracting it into a variable is adding another layer of indirection to figure out what the project needs.

Reply:
Seems like most project use it what if we made it included by default and only opted-out where a project didn't want to use it?

Reply:
IMO we are in the worst place right now.
One part of the shared files is included by properties, the other part by direct  `Compile` calls.

We should move all projects in one direction. When we have agreement, I am fine to extend contribution guide :) .

Reply:
> IMO we are in the worst place right now.
One part of the shared files is included by properties, the other part by direct Compile calls.

I see value in extracting **multiple** Compile includes into a property. There are a bunch of projects that use a common set of Compile includes. But extracting a **single** compile include into a property doesn't make sense to me. If that's what we want to do then, where do we stop? Why don't we create a property for every single PackageReference and ProjectReference as well?

> We should move all projects in one direction. When we have agreement, I am fine to extend contribution guide :) .

If we want to be consistent, I would be in favor of removing these properties altogether and have each project explicitly add the individual Compile Include they need.